### ISSUE TITLE
Collections: Create discovery partial of the new/edit form

### ISSUE THREAD

**Issue by elrayle** [samvera/hyrax#1379]
### Descriptive summary

Add code to implement the Discovery tab partial for the new/edit form views for collections.  This partial is shared by both the new and edit form views.

### Rationale

See model issue for rationale.

### Implementation

**Files:**
 * Edit [app/views/hyrax/collections/_form.html.erb](https://github.com/samvera/hyrax/blob/master/app/views/hyrax/collections/_form.html.erb)
 * Refactor-rename [app/views/hyrax/collections/_form_permission.html.erb](https://github.com/samvera/hyrax/blob/master/app/views/hyrax/collections/_form_permission.html.erb) to _form_discovery.html.erb≥
 

#### Expected user interactions

Modify the tab in the form by... 
- [x] rename the view partial file for setting discoverability of a collection from `_form_permission.html.erb` to `_form_discovery.html.erb`.
- [x] edit _form.html.erb and change the partial being rendered from `form_permission` to `form_discovery`.html.erb`.
- [x] change the translation for the tab label from `Visibility` to `Discovery`

The actual content of the partial is unchanged.

Exact text for the page is in the mockup below.

#### SPECIAL NOTE on Admin Sets:

For the initial implementaiton, the Admin Set underlying code will not change.  But to a user, the Admin Set will appear to be a collection.

To make the illusion happen...
* QUESTION: Are admin sets discoverable?  Can you turn discovery on/off?  To be fully consistent, a site should be able to configure this.  If it is not an option and/or not settable via the UI, create a new low priority issue that we will get to if there is time.

### Mockup

![collection-new-edit-discovery](https://user-images.githubusercontent.com/101482/28546327-24530410-707f-11e7-8213-854cb30a7a33.png)

### Related work

| Collections feature | Issue |
| ------------------------ | ----- |
| Routes                   | #1380 |
| Controller               | # |
| Model                    | #1383 |
| View - Index            | #1417 |
| View - Show - Admin      | #1427 |
| View - Show - Landing    | #1428 |
| View - New/Edit          | #1374 |
| Partial -  Description   | #1375 |
| Partial -  Branding      | #1376 |
| Partial -  Relationships | #1377 |
| Partial -  Sharing       | #1378 |
| Partial -  Discovery     | (this issue) |

**Comment by elrayle** [comment]
It may not be obvious from the mockup that the text Open Access, Institution, and Private are bold, while the description of each of those is not bold. For collection types, I handled this by having a translation for the label (e.g. Open Access) and the description (e.g. Visible to all). Then they can be styled differently in the view.

Example from [collection types settings partial.](https://github.com/samvera/hyrax/blob/collections-sprint/config/locales/hyrax.en.yml#L177-L179)..
```
            nestable:
              label:         "NESTABLE"
              description:   "Allow collections of this type to be nested (a collection can contain other collections)"
```
I merged PR #1448 but kept this issue open to address this minor UI issue.
